aspnet/mvc/views/forms/sample/final/Controllers/DemoController.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Microsoft.AspNet.Mvc;
+
+namespace FormsTagHelper.Controllers
+{
+    // This controller is used only to demonstrate working with forms.
+
+    public class DemoController : Controller
+    {       
+        
+        public IActionResult Index(string ID)
+        {
+            ViewData["Message"] = ID;
+            return View();
+        }
+
+        public IActionResult RegisterTextArea()
+        {
+            return View();
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult RegisterTextArea(DescriptionViewModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                string success = "HttpPost Register success " + model.Description;
```

Comment:
FYI if this were an aspnet code review, would be recommending `var` throughout this sample. Up to you whether the increased brevity is a good thing.